added GitLab repo scanning support

anubhav06 opened the pull request:
Signed-off-by: Anubhav Gupta <mail.anubhav06@gmail.com>


## Describe your changes
Added support for scanning remote repository hosted on GitLab
This PR is linked with "[added GitLab API support](https://github.com/kubescape/go-git-url/pull/4)" PR, and that needs to be merged before, in order for this to work.

## Screenshots - If Any (Optional)

## This PR fixes:

* Resolved #953 

## Checklist before requesting a review
<!-- put an [x] in the box to get it checked -->

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] New and existing unit tests pass locally with my changes

dwertent commented:
Hi @anubhav06 , did you test this?

I ran (public repo):
```
kubescape scan https://gitlab.com/dwertent/testing.git
```

And I see this log in my output:
```
[error] [lstat /<path>/https:/gitlab.com/dwertent/: no such file or directory]
```

This means you need to add support for cloning GitLab repositories as described in the relate issue.

anubhav06 commented:
Hello @dwertent . Yes I did this it.
It does work for me. When I ran the same command, `kubescape scan https://gitlab.com/dwertent/testing.git`, I get these results:  
  

![image](https://user-images.githubusercontent.com/43821031/207001895-e4bd5ddd-c51d-4b90-9321-fd5c80234bec.png)

The rendering of Helm chart templates error we see, I don't think it has anything to do with GitLab scanning support, but I can take a look at that.

dwertent commented:
Are you sure you pushed all of your changes? It is not working for me...

anubhav06 commented:
For testing it locally, did you update KubeScape to use the updated [`kubescape/go-git-url`](https://github.com/kubescape/go-git-url) repo in which I have added support for the GitLab API [**in this PR**](https://github.com/kubescape/go-git-url/pull/4). 
I think you are using the present version of `kubescape/go-git-url` and not referencing it to my PR.

Let me once list the process I'm following to make it run locally, and see if it helps:

1. In the `go.mod` file of this KubeScape repo, I added the line: 
`replace github.com/kubescape/go-git-url => ../go-git-url` in the end, for using my local copy of `go-git-url` (in which I've added support for GitLab API), instead of the one hosted on GitHub.
2. Ran `go mod tidy`
3. Ran `make build`

After this KubeScape refers to my local `go-git-url` while running scans
